examples/javascript/pkce-browser/index.html

```diff
@@ -0,0 +1,116 @@
+            </h2>
+        </div>
+    </header>
+
+    <!-- Example description and UI -->
+    <section class="container main">
+        <p>This example shows how to use PKCE in the browser</p>
+        <div id="pre-auth-section" style="display:none;">
+            <button onClick="doAuth()">Start PKCE Auth Flow</button>
+
+            <p class="info">Once authenticated, it will use the access token to list the files in your root directory.</p>
+        </div>
+
+        <div id="authed-section" style="display:none;">
+            <p>You have successfully authenticated. Below are the contents of your root directory. They were fetched using the SDK and access token.</p>
+            <ul id="files"></ul>
+        </div>
+    </section>
+
+    <!-- Scripts to run example -->
+    <script>
+        var REDIRECT_URI = 'http://localhost:8080/pkce-browser';
+        var CLIENT_ID = '42zjexze6mfpf7x';
+        var dbxAuth = new Dropbox.DropboxAuth({
+            clientId: CLIENT_ID,
+        });
+
+        // Parses the url and gets the access token if it is in the urls hash
+        function getCodeFromUrl() {
+            return utils.parseQueryString(window.location.search).code;
+        }
+
+        // If the user was just redirected from authenticating, the urls hash will
+        // contain the access token.
+        function hasRedirectedFromAuth() {
+            return !!getCodeFromUrl();
+        }
+
+        // Render a list of items to #files
+        function renderItems(items) {
+            var filesContainer = document.getElementById('files');
+            items.forEach(function(item) {
+                var li = document.createElement('li');
+                li.innerHTML = item.name;
+                filesContainer.appendChild(li);
+            });
+        }
+
+        // This example keeps both the authenticate and non-authenticated setions
+        // in the DOM and uses this function to show/hide the correct section.
+        function showPageSection(elementId) {
+            document.getElementById(elementId).style.display = 'block';
+        }
+
+        function doAuth() {
+            dbxAuth.getAuthenticationUrl(REDIRECT_URI, undefined, 'code', 'offline', undefined, undefined, true)
+                .then(authUrl => {
+                    window.sessionStorage.clear();
+                    window.sessionStorage.setItem("codeVerifier", dbxAuth.codeVerifier);
+                    window.location.href = authUrl;
+                })
+                .catch((error) => console.error(error));
+        };
+
+        if (hasRedirectedFromAuth()) {
+            showPageSection('authed-section');
+            dbxAuth.setPkceCodeVerifierAndGenerateChallenge(window.sessionStorage.getItem('codeVerifier'));
+            dbxAuth.getAccessTokenFromCode(REDIRECT_URI, getCodeFromUrl())
+                .then((response) => {
+                    dbxAuth.setAccessToken(response.result.access_token);
```

Comment:
Could `response.result` be undefined?

Reply:
I guess hypothetically it could be but this is meant to be a simple example showing users how to use the endpoints not be a fully robust example - notice error handling is minimal and even edge case handling isn't here

Reply:
Makes sense!

Reply:
Where is `setPkceCodeVerifierAndGenerateChallenge` defined? It's not a function when you `import { DropboxAuth } from 'dropbox'`

Reply:
@jonahlibrach Thanks for the note. This is open as https://github.com/dropbox/dropbox-sdk-js/issues/596 .

Reply:
@greg-db how do I get this working? I assume that it works for _**someone**_ since it was merged to main?

Reply:
We've merged in [the correct version](https://github.com/dropbox/dropbox-sdk-js/blob/main/examples/javascript/pkce-browser/index.html) now. Please give that a try. Thanks!